### DISCUSSION TITLE
Deployment with managed database service (AWS RDS)

### DISCUSSION THREAD

**Discussion by mattigrthr** [bytesandbrains/h3-pg#68] [Q&A] [answered]
Did somebody manage to deploy h3-pg with a managed database service? I got a running docker image and it works fine locally but now we need to deploy it. I know I could just create a docker image on a server but then I would lose the advantages such as auto-scaling or managed backups of a managed database service such as AWS RDS.

This is the dockerfile I'm currently using locally. Is there a way to do the same installation steps for a managed service?

```
FROM kartoza/postgis:14-3.1

# Install h3-pg
RUN apt-get update && apt-get install -y cmake gcc git libtool make python3 python3-pip
RUN pip3 install --no-cache --upgrade pip setuptools && pip3 install pgxnclient
RUN pgxn install h3

# Initialize tables script
RUN mkdir -p /docker-entrypoint-initdb.d
COPY ./core/database/importer/sql/create_tables.sql /docker-entrypoint-initdb.d/create_tables.sql
```

AWS RDS specifically has a [list of supported extensions](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts.General.FeatureSupport.Extensions) that doesn't include h3-pg. I found this [old thread on stackexchange](https://dba.stackexchange.com/questions/111477/amazon-rds-postgresql-adding-new-extensions) which says it's not possible to install any C-related extensions on RDS. So right now it seems to be not possible. :(

How do you use h3-pg in a production environment? Would appreciate any tips! :)

**Comment by zachasme** [comment]
We are using it with our own machines. Unfortunately I do not know if is is possible, but I too am very interested to hear if there are ways to deploy to a managed database service :-)

**Comment by sondrelg** [comment]
Did you ever manage to make it work @mattigrthr?

**Reply by mattigrthr** [reply]
Unfortunately not, @sondrelg.  Just used the above Docker image on an EC2 instance.

**Comment by Komzpa** [comment, marked as answer]
https://docs.aws.amazon.com/AmazonRDS/latest/PostgreSQLReleaseNotes/postgresql-extensions.html

h3-pg is now supported on Amazon RDS

**Reply by zachasme** [reply]
Amazing! 😮🙏
